crates/starknet/src/starknet/mod.rs

```diff
@@ -467,6 +576,23 @@ impl Starknet {
         add_invoke_transaction::add_invoke_transcation_v1(self, invoke_transaction)
     }
 
+    // save starknet transactions to file
+    pub fn dump_transactions(&self) {
+        // TODO: attach in OnTransaction mode or dump all??
+        match &self.config.dump_path {
+            Some(path) => {
+                let starknet_dump = Some(
+                    serde_json::to_string(&self.transactions)
+                        .expect("Failed to serialize starknet transactions"),
+                );
+                let encoded: Vec<u8> = bincode::serialize(&starknet_dump)
+                    .expect("Failed to encode starknet transactions");
+                fs::write(Path::new(&path), encoded).expect("Failed to save starknet transactions");
+            }
+            None => {} // TODO: Error here?
```

Comment:
Yes, shall return an error. I we're here that dump action exist

Reply:
fixed in 36c55738b8ce64d93912e10c4780b1f2424f1e67 is that ok?

Reply:
Discussed in dm.

Reply:
Fixed in 27ec8cc63261fb9430c873a0adda2ceebf38f005